[LibWebRTC] Fix Linux, GStreamer and unified build

blino opened the pull request:
#### 80245993fa5ad38a289984f5a377c78384fcaae7
<pre>
[LibWebRTC] Fix Linux, GStreamer and unified build
<a href="https://bugs.webkit.org/show_bug.cgi?id=286816">https://bugs.webkit.org/show_bug.cgi?id=286816</a>

Reviewed by Philippe Normand.

* Source/ThirdParty/libwebrtc/CMakeLists.txt:
  - Remove test support files requiring gtest headers, not exposed in CMake build
    (introduced in libwebrtc M132 update, 287962@main)

* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCMediaEndpoint.cpp:
(WebCore::LibWebRTCMediaEndpoint::createStatsCollector):
  - Do not move rtcReport to avoid const qualified error (introduced in 287147@main)

* Source/WebCore/platform/mediastream/libwebrtc/gstreamer/GStreamerVideoEncoderFactory.cpp:
(WebCore::GStreamerVideoEncoder::makeElement):
  - Use ASCIILiteral to avoid error from missing span(const gchar*) (from 279038@main)

* Source/WebKit/WebProcess/Network/webrtc/WebRTCMonitor.cpp:
  - Include missing LibWebRTCNetwork.h for ref/deref methods

Canonical link: <a href="https://commits.webkit.org/289776@main">https://commits.webkit.org/289776@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3fd0ccac8264cd633147f9ca6d985c1ffd687c48

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/87552 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/7067 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/41930 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/92415 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/38296 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/89603 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/7448 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/15235 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/67623 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/25374 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/90554 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/5692 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/79234 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/47977 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/5478 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/33627 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/37409 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/75885 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/34492 "Passed tests") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/94302 "") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/14720 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/10801 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/94302 "") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/14974 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/75083 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/94302 "") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18687 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/20061 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/18488 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/7671 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/14736 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/20037 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/14480 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/17924 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/16262 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

blino commented on Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCMediaEndpoint.cpp:
```diff
@@ -279,7 +279,7 @@ rtc::scoped_refptr<LibWebRTCStatsCollector> LibWebRTCMediaEndpoint::createStatsC
         Ref peerConnectionBackend = protectedThis->m_peerConnectionBackend.get();
         Ref peerConnection = peerConnectionBackend->connection();
-        peerConnection->queueTaskKeepingObjectAlive(peerConnection.get(), TaskSource::Networking, [promise = WTFMove(promise), rtcReport = WTFMove(rtcReport)] {
+        peerConnection->queueTaskKeepingObjectAlive(peerConnection.get(), TaskSource::Networking, [promise = WTFMove(promise), rtcReport] {
```
Not sure about this one.
The build error with the Linux Flatpak toolchain is:
```
/app/webkit/WebKitBuild/WPE/Debug/WTF/Headers/wtf/StdLibExtras.h: In instantiation of ‘constexpr typename std::remove_reference<_Arg>::type&& std::move(T&&) [with WTF::CheckMoveParameterTag <anonymous> = WTF::CheckMoveParameter; T = const webrtc::scoped_refptr<const webrtc::RTCStatsReport>&; typename remove_reference<_Arg>::type = const webrtc::scoped_refptr<const webrtc::RTCStatsReport>]’:
/app/webkit/Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCMediaEndpoint.cpp:282:140:   required from here
/app/webkit/WebKitBuild/WPE/Debug/WTF/Headers/wtf/StdLibExtras.h:817:51: error: static assertion failed: T is const qualified.
  817 |     static_assert(!is_const<NonRefQualifiedType>::value, "T is const qualified.");
      |                                                   ^~~~~
/app/webkit/WebKitBuild/WPE/Debug/WTF/Headers/wtf/StdLibExtras.h:817:51: note: ‘!(bool)std::integral_constant<bool, true>::value’ evaluates to false
```

philn replied:
What is the error this change intends to fix?

blino replied:
This is in the comment above, any more details needed?

philn replied:
I didn't check the red gtk ews errors, i thought that error you shared was the one from ews.

philn replied:
This is fine i think because `createReport` takes a const ref anyway. I don't think moving the report to the closure is much relevant.

blino replied:
Thanks for the review!

blino committed:
[LibWebRTC] Fix Linux, GStreamer and unified build
https://bugs.webkit.org/show_bug.cgi?id=286816

Reviewed by Philippe Normand.

* Source/ThirdParty/libwebrtc/CMakeLists.txt:
  - Remove test support files requiring gtest headers, not exposed in CMake build
    (introduced in libwebrtc M132 update, 287962@main)

* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCMediaEndpoint.cpp:
(WebCore::LibWebRTCMediaEndpoint::createStatsCollector):
  - Do not move rtcReport to avoid const qualified error (introduced in 287147@main)

* Source/WebCore/platform/mediastream/libwebrtc/gstreamer/GStreamerVideoEncoderFactory.cpp:
(WebCore::GStreamerVideoEncoder::makeElement):
  - Use ASCIILiteral to avoid error from missing span(const gchar*) (from 279038@main)

* Source/WebKit/WebProcess/Network/webrtc/WebRTCMonitor.cpp:
  - Include missing LibWebRTCNetwork.h for ref/deref methods

Canonical link: https://commits.webkit.org/289776@main
